### PR TITLE
feat(go-import): support branch selection for go vanity urls

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

**Comment by flynnduism** [comment]
@adamreese I'm unclear on what this does?

Is the new `content/code/` addition to serve as an example? Or this PR this adding metadata to support versioning somehow?

**Comment by bacongobbler** [comment]
> I'm unclear on what this does?

This is a feature of `go get`. Essentially, by hosting a page on a domain with a particular `<meta>` tag that points to a repository, `go get` can use that domain instead of the repository. In this example, `go get helm.sh/helm` is the same as `go get github.com/helm/helm`, with a few subtle differences.

The change in this PR makes it such that `go get helm.sh/helm` fetches from the dev-v3 branch instead of the `master` branch.

**Comment by bacongobbler** [comment]
More information on remote import paths:

https://golang.org/cmd/go/#hdr-Remote_import_paths
https://blog.bramp.net/post/2017/10/02/vanity-go-import-paths/

File: themes/helm/layouts/code/single.html [hunk 1]
```diff
@@ -3,14 +3,13 @@
 
 <head>
   <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
-  <meta name="go-import" content="https://helm.sh{{substr .RelPermalink 0 -1}} git {{.Params.vanity}}">
-  <meta name="go-source" content="https://helm.sh{{substr .RelPermalink 0 -1}} {{.Params.vanity}} {{.Params.vanity}}/tree/master{/dir} {{.Params.vanity}}/blob/master{/dir}/{file}#L{line}">
-
-  <meta http-equiv="refresh" content="0; url={{.Params.vanity}}">
+  <meta name="go-import" content="https://helm.sh/{{.Params.Name}} git {{.Params.RepoURL}}">
```
Comment (bacongobbler) [inline themes/helm/layouts/code/single.html]: It looks like the parameters are camel-cased, but the params in helm.md weren't changed. Have you tested and confirmed this works?

Comment (adamreese) [reply themes/helm/layouts/code/single.html]:
```
❯ curl -L http://localhost:1313/helm
<!DOCTYPE html>
<html lang="en-us">

<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
  <meta name="go-import" content="https://helm.sh/helm git https://github.com/helm/helm">
  <meta name="go-source" content="https://helm.sh/helm git https://github.com/helm/helm https://github.com/helm/helm/tree/dev-v3{/dir} https://github.com/helm/helm/blob/dev-v3{/dir}/{file}#L{line}">
  <meta http-equiv="refresh" content="0; url=https://github.com/helm/helm">
</head>

<body>
  Nothing to see here; <a href="https://github.com/helm/helm">move along</a>.
<script data-no-instant>document.write('<script src="/livereload.js?port=1313&mindelay=10"></' + 'script>')</script></body>

</html>
```

**Pull request merged**